packages/node_modules/nav-frontend-stegindikator/src/stegindikator.tsx

```diff
@@ -0,0 +1,211 @@
+    getInnerStep() {
+        const labelTemplate = (<div className="stegindikator__steg-label">{this.props.label}</div>);
+        const label = (this.props.visLabel) ? labelTemplate : undefined;
+
+        if (typeof this.props.onClick === 'function' && !this.props.aktiv) {
+            return (
+                <button
+                    className={innerCls(this.props)}
+                    title={this.props.label}
+                    onClick={this.props.onClick}
+                    disabled={this.props.disabled}
+                >
+                    <div className="stegindikator__steg-num">{this.props.index}</div>
+                    {label}
+                </button>
+            );
+        }
+
+        return (
+            <div className={innerCls(this.props)} title={this.props.label}>
+                <div className="stegindikator__steg-num">{this.props.index}</div>
+                {label}
+            </div>
+        );
+    }
+
+    render() {
+        const domProps = omit(
+            this.props,
+            'label',
+            'aktiv',
+            'ferdig',
+            'visLabel',
+            'index',
+            'disabled',
+            'onClick'
+        );
+
+        return (
+            <li className={stegCls(this.props)} {...domProps}>
+                {this.getInnerStep()}
+            </li>
+        );
+    }
+}
+
+class NavFrontendStegindikator extends React.Component<NavFrontendStegindikatorProps, NavFrontendStegindikatorState> {
+    static defaultProps: Partial<NavFrontendStegindikatorProps> = {
+        visLabel: false,
+        kompakt: false,
+        aktivtSteg: 1,
+        ferdigSteg: [],
+        autoResponsiv: false
+    };
+
+    private list: HTMLOListElement;
+
+    constructor(props: NavFrontendStegindikatorProps) {
+        super(props);
+        this.state = {
+            visLabel: props.visLabel,
+            kompakt: props.kompakt
+        };
+
+        if (props.autoResponsiv) {
+            window.addEventListener('resize', this.checkSize);
+        }
+    }
+
+    componentDidMount() {
+        if (this.props.autoResponsiv) {
+            this.checkSize();
+        }
+    }
+
+    componentWillReceiveProps(nextProps: NavFrontendStegindikatorProps) {
+        if (!this.props.autoResponsiv && nextProps.autoResponsiv) {
+            window.addEventListener('resize', this.checkSize);
+        }
+
+        if (nextProps.autoResponsiv) {
+            this.setState({
+                visLabel: (this.canShowLabel()) ? nextProps.visLabel : false,
+                kompakt: nextProps.kompakt
+            });
+        } else {
+            this.setState({
+                visLabel: nextProps.visLabel,
+                kompakt: nextProps.kompakt
+            });
+        }
+    }
+
+    componentWillUnmount() {
+        window.removeEventListener('resize', this.checkSize);
+    }
+
+    getDimensions() {
+        let remSize = parseFloat(String(getComputedStyle(document.documentElement).fontSize));
```

Comment:
Hva er formålet med `getDimensions` og all matematikken her?

Reply:
Ettersom det er umulig å forutsi hvor mange steg en gitt stegindikator vil få, så er det ikke mulig å skrive noe ut-av-boksen CSS-regler for hvordan komponenten skal være responsiv. La derfor opp til at bruker kan angi en `autoResponsiv`-prop som oppretter en event listener som på mount og `resize` kjører en sjekk på total og tilgjengelig bredde, og gjør komponenten mer og mer kompakt etterhvert som tilgjengelig bredde minker. `getDimensions` brukes her til å finne ut komponentens bredde.